core/graph/builders/unix_socket_client.py

```
@@ -0,0 +1,122 @@
+from arango.http import HTTPClient
+from arango.response import Response
+from urllib.parse import urlparse, urlunparse
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class UnixSocketHTTPClient(HTTPClient):
+    """HTTP client that uses Unix domain sockets instead of TCP."""
+    
+    def __init__(self, unix_socket_path='/tmp/arangodb.sock'):
```

Ruff (0.12.2)

Probable insecure usage of temporary file or directory: "/tmp/arangodb.sock"

(S108)